File: tools/orchestrator/src/util/fsUtil.ts
import fs from "node:fs/promises";

export async function ensureDir(dirPath: string): Promise<void> {
  await fs.mkdir(dirPath, { recursive: true });
}

export async function exists(p: string): Promise<boolean> {
  try {
    await fs.access(p);
    return true;
  } catch {
    return false;
  }
}

File: backend/src/resume_orchestrator/util/fs_util.py
from __future__ import annotations

from pathlib import Path


def ensure_dir(dir_path: str | Path) -> Path:
    p = Path(dir_path)
    p.mkdir(parents=True, exist_ok=True)
    return p


def exists(p: str | Path) -> bool:
    return Path(p).exists()
